### ISSUE TITLE
Update frontend to use `allowNull` feature of `PUT /item/<id>/metadata` endpoint

### ISSUE THREAD

**Issue by ronichoudhury** [girder/girder#1863]
Now that we are able to store actual `null` values in metadata, we should allow it to happen via the frontend.

What tasks would this entail? I believe this list will cover it:
- set `allowNull` to true when the metadata update call happens (i.e., when the "save" button is clicked)
- exclusively use the new `DELETE /item/<id>/metadata` endpoint to remove metadata entries that disappear from the metadata record when the "save" button is clicked

**Comment by zachmullen** [comment]
I've added the "bug" label here, even though this was initially a conscious design decision. I think it's counterintuitive when you add a JSON metadata field and write "null" in the value, and then when you save the field and reload, it just disappears altogether.

**Comment by ronichoudhury** [comment]
Agreed, especially because there's already a correct way to delete a metadata field in the UI.
